pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -177,8 +178,14 @@ func (c *ReverseExpandQuery) Execute(
 	resultChan chan<- *ReverseExpandResult,
 	resolutionMetadata *ResolutionMetadata,
 ) error {
-	defer close(resultChan)
-	return c.execute(ctx, req, resultChan, nil, resolutionMetadata)
+	err := c.execute(ctx, req, resultChan, nil, resolutionMetadata)
+	if err != nil {
+		resultChan <- &ReverseExpandResult{Err: err}
+	} else {
+		close(resultChan)
+	}
+
+	return err
```

Comment:
The error case looks very odd... we send `err` through the channel _and_ we return it synchronously at the very least we need to document this so that clients (right now, only ListObjects) know how to handle it.

Reply:
I've reworked the error handling to produce an error on the channel and close the channel if Execute returns without any error. Let me know what you think 👍